File: examples/canonical-terminal/python-typer-textual/tests/smoke/test_cli_smoke.py
import json
from pathlib import Path

from typer.testing import CliRunner

from taskflow.cli.main import app
from taskflow.core.loader import default_fixtures_path


runner = CliRunner()


def _base_args() -> list[str]:
    return ["--fixtures-path", str(default_fixtures_path())]


def test_list_table_output() -> None:
    result = runner.invoke(app, ["list", *_base_args()])
    assert result.exit_code == 0
    assert "## BEGIN_JOBS ##" in result.stdout
    assert "## END_JOBS ##" in result.stdout
    assert "build-frontend" in result.stdout


def test_list_json_output() -> None:
    result = runner.invoke(app, ["list", *_base_args(), "--output", "json"])
    assert result.exit_code == 0
    payload = json.loads(result.stdout)
    assert len(payload) == 20


def test_list_filter_by_status() -> None:
    result = runner.invoke(app, ["list", *_base_args(), "--status", "done", "--output", "json"])
    assert result.exit_code == 0
    payload = json.loads(result.stdout)
    assert payload
    assert all(job["status"] == "done" for job in payload)


def test_inspect_job() -> None:
    result = runner.invoke(app, ["inspect", "job-001", *_base_args(), "--output", "json"])
    assert result.exit_code == 0
    payload = json.loads(result.stdout)
    assert payload["id"] == "job-001"


def test_stats_json() -> None:
    result = runner.invoke(app, ["stats", *_base_args(), "--output", "json"])
    assert result.exit_code == 0
    payload = json.loads(result.stdout)
    assert payload["total"] == 20


def test_watch_no_tui() -> None:
    result = runner.invoke(app, ["watch", *_base_args(), "--no-tui"])
    assert result.exit_code == 0
    assert "## BEGIN_JOBS ##" in result.stdout


def test_list_missing_fixtures() -> None:
    missing = Path(default_fixtures_path()) / "missing-dir"
    result = runner.invoke(app, ["list", "--fixtures-path", str(missing)])
    assert result.exit_code == 1

File: examples/canonical-terminal/ruby-thor-tty/tests/smoke/test_cli_smoke.rb
# frozen_string_literal: true

require "json"
require "minitest/autorun"
require "open3"
require "pathname"

class TaskflowCliSmokeTest < Minitest::Test
  def setup
    @root = Pathname(__dir__).join("../..").expand_path
    @fixtures = @root.join("../fixtures").expand_path
    @bin = @root.join("bin/taskflow").to_s
    @env = { "TASKFLOW_FIXTURES_PATH" => @fixtures.to_s }
  end

  def test_list_table
    stdout, stderr, status = run_cli("list")
    assert status.success?, stderr
    assert_includes stdout, "## BEGIN_JOBS ##"
    assert_includes stdout, "job-001"
  end

  def test_list_json
    stdout, stderr, status = run_cli("list", "--output", "json")
    assert status.success?, stderr
    payload = JSON.parse(stdout)
    assert_equal 20, payload.length
  end

  def test_filter
    stdout, stderr, status = run_cli("list", "--status", "done", "--output", "json")
    assert status.success?, stderr
    payload = JSON.parse(stdout)
    assert payload.all? { |job| job["status"] == "done" }
  end

  def test_inspect
    stdout, stderr, status = run_cli("inspect", "job-001", "--output", "json")
    assert status.success?, stderr
    payload = JSON.parse(stdout)
    assert_equal "job-001", payload["id"]
  end

  def test_stats
    stdout, stderr, status = run_cli("stats", "--output", "json")
    assert status.success?, stderr
    payload = JSON.parse(stdout)
    assert_equal 20, payload["total"]
  end

  def test_watch_batch
    stdout, stderr, status = run_cli("watch", "--no-interactive")
    assert status.success?, stderr
    assert_includes stdout, "## BEGIN_JOBS ##"
  end

  def test_missing
    stdout, stderr, status = run_cli("list", "--fixtures-path", @root.join("missing").to_s)
    refute status.success?
    assert_includes stdout + stderr, "fixtures path does not exist"
  end

  private

  def run_cli(*args)
    Open3.capture3(@env, "ruby", @bin, *args, chdir: @root.to_s)
  end
end
